tests/test_cookiecutter_generation.py

```diff
@@ -157,3 +162,11 @@ def test_invalid_slug(cookies, context, slug):
 
     assert result.exit_code != 0
     assert isinstance(result.exception, FailedHookException)
+
+
+def test_no_whitenoise_and_no_cloud_provider(cookies, context):
+    """It should not generate project if neither whitenoise or cloud provider are set"""
+    context.update({"use_whitenoise": "n", "cloud_provider": "None"})
+    result = cookies.bake(extra_context=context)
+
+    assert result.exit_code == 1
```

Comment:
Looks like it's `-1` for some reason... I suppose it's good enough to check that it's non-zero, and that the exception type is what we want. We don't really need a specific number:

```suggestion
    assert result.exit_code != 0
    assert isinstance(result.exception, FailedHookException)
```